[Inductor] [Triton] Capture Timeout errors without crashing the job

njriasan opened the pull request:
Summary:
Opts to capture timeout errors during compilation without forcing process failure. Useful to avoid hangs in MAST jobs.

We may want to consider a configuration option for this to avoid wasted compute by never pruning bad config options.

Test Plan: Tested with local model reproducers.

Differential Revision: D87866423




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

PaulZhang12 reviewed (approved):
Overall, looks good to me!

njriasan commented on torch/_inductor/select_algorithm.py:
```diff
@@ -3241,44 +3241,71 @@ def wait_on_futures():
+                        from torch._inductor.codegen.cuda.cuda_kernel import (
+                            CUDATemplateCaller,
                         )
-                        futures[future].mark_failed()
```
Looks like a quality issue in the old code. Seems like the if/else should just be deleted. Good find!

njriasan committed:
[Inductor] [Triton] Capture Timeout errors without crashing the job (#169064)

Summary:

Opts to capture timeout errors during compilation without forcing process failure. Useful to avoid hangs in MAST jobs.

We may want to consider a configuration option for this to avoid wasted compute by never pruning bad config options.

Test Plan: Tested with local model reproducers.

Reviewed By: PaulZhang12

Differential Revision: D87866423
